internal/pods/schedule.go

```diff
@@ -271,3 +286,37 @@ func addNodeSelectorTerm(nodeSelTerms []v1.NodeSelectorTerm, label string,
 		},
 	}})
 }
+
+func checkNSLabels(ns *v1.Namespace) {
```

Comment:
```suggestion
func checkNSLabels(config *Config) error {
	ns, err := config.ClientSet.CoreV1().Namespaces().Get(context.TODO(), config.Namespace, metav1.GetOptions{})
	if err != nil {
		return errors.Wrap(err, fmt.Sprintf("error fetching %s namespace", config.Namespace))
	}

```

Less repetition, plus namespace object isn't used outside so its OK to fetch it just for this function